packages/trader/src/App/Components/Elements/ContractDrawer/contract-drawer.jsx

```diff
@@ -13,48 +13,26 @@ import { getEndTime } from 'Stores/Modules/Contract/Helpers/logic';
 import ContractDrawerCard from './contract-drawer-card.jsx';
 import { SwipeableContractAudit } from './swipeable-components.jsx';
 
-class ContractDrawer extends React.Component {
-    contract_drawer_ref = React.createRef();
-    contract_drawer_card_ref = React.createRef();
-
-    state = {
-        is_shade_on: false,
-        should_show_contract_audit: false,
-    };
-
-    handleShade = shade => {
-        this.setState({ is_shade_on: shade });
-    };
-
-    toggleContractAuditDrawer = () => {
-        this.setState(state => ({ should_show_contract_audit: !state.should_show_contract_audit }));
-    };
-
-    onSwipedDown = () => this.setState({ should_show_contract_audit: false });
-    onSwipedUp = () => this.setState({ should_show_contract_audit: true });
-
-    get is_collapsed() {
-        return this.state.should_show_contract_audit;
-    }
-
-    getBodyContent() {
-        const { currency, exit_tick_display_value, is_sold } = this.props.contract_info;
-
-        const {
-            contract_info,
-            contract_update,
-            contract_update_history,
-            is_mobile,
-            is_sell_requested,
-            is_dark_theme,
-            is_multiplier,
-            onClickCancel,
-            onClickSell,
-            server_time,
-            status,
-            toggleHistoryTab,
-        } = this.props;
-
+const ContractDrawer = ({
+    contract_info,
+    contract_update,
+    contract_update_history,
+    is_mobile,
+    is_sell_requested,
+    is_dark_theme,
+    is_multiplier,
+    onClickCancel,
+    onClickSell,
+    server_time,
+    status,
+    toggleHistoryTab,
+}) => {
+    const { currency, exit_tick_display_value, is_sold } = contract_info;
+    const contract_drawer_ref = React.createRef();
+    const contract_drawer_card_ref = React.createRef();
```

Comment:
@amirghadimi-fs any reason not using `useRef`?

Reply:
These are old codes I didn’t touch them. Do you suggest to use `useRef`?

Reply:
@amirghadimi-fs it seems `createRef` create a ref each time, but `useRef` re-use it if it already created. better to use `useRef`. thanks

Reply:
Sure I'll change it.